mmdet/models/losses/gaussian_focal_loss.py

```diff
@@ -0,0 +1,53 @@
+import torch.nn as nn
+
+from ..builder import LOSSES
+
+
+def gaussian_focal_loss(pred, target, weight=None, alpha=2.0, gamma=4.0):
+    eps = 1e-12
+    pos_weights = target.eq(1)
+    neg_weights = (1 - target).pow(gamma)
+    pos_loss = -(pred + eps).log() * (1 - pred).pow(alpha) * pos_weights
+    neg_loss = -(1 - pred + eps).log() * pred.pow(alpha) * neg_weights
+
+    if weight is not None:
+        pos_loss *= weight
+        neg_loss *= weight
+
+    pos_num = pos_weights.sum()
+    if pos_num < 1:
+        loss = neg_loss.sum()
+    else:
+        loss = (pos_loss + neg_loss).sum() / pos_num
+    return loss
+
+
+@LOSSES.register_module()
+class GaussianFocalLoss(nn.Module):
```

Comment:
The gaussian is encoded in `target`. 
So `GaussianFocalLoss` may not be precise.